src/features/wallet/helpers/collectRewards.ts

```diff
@@ -32,46 +33,84 @@ export default function collectRewards(
+
+      data.forEach((datum) => {
+        if (datum.roundId === roundId) {
+          const pendingRequest = {} as PendingRequestRetrieveReward;
+          postData.roundId = datum.roundId;
+          postData.voterAddress = datum.address;
+          pendingRequest.ancillaryData = datum.ancillaryData
+            ? datum.ancillaryData
+            : "0x";
+          pendingRequest.identifier = ethers.utils.toUtf8Bytes(
+            datum.identifier
+          );
+          pendingRequest.time = datum.time;
+          pendingRequestData.push(pendingRequest);
+        }
+      });
+
+      postData.pendingRequests = pendingRequestData;
+      mcr.target = contract.address;
+      mcr.callData = votingInterface.encodeFunctionData(
+        "retrieveRewards(address,uint256,(bytes32,uint256,bytes)[])",
+        [postData.voterAddress, postData.roundId, postData.pendingRequests]
+      );
 
-    data.forEach((datum) => {
-      if (datum.roundId === roundId) {
-        const pendingRequest = {} as PendingRequestRetrieveReward;
-        postData.roundId = datum.roundId;
-        postData.voterAddress = datum.address;
-        pendingRequest.ancillaryData = datum.ancillaryData
-          ? datum.ancillaryData
-          : "0x";
-        pendingRequest.identifier = ethers.utils.toUtf8Bytes(datum.identifier);
-        pendingRequest.time = datum.time;
-        pendingRequestData.push(pendingRequest);
-      }
+      multicallCollectRewards.push(mcr);
     });
 
-    postData.pendingRequests = pendingRequestData;
-    mcr.target = contract.address;
-    mcr.callData = votingInterface.encodeFunctionData(
-      "retrieveRewards(address,uint256,(bytes32,uint256,bytes)[])",
-      [postData.voterAddress, postData.roundId, postData.pendingRequests]
+    return (
+      multicallContract.functions["aggregate((address,bytes)[])"](
+        multicallCollectRewards
+      )
+        // wait for at least 1 block conf.
+        .then((tx) => tx.wait(1))
+        .then((conf: any) => {
+          setAvailableRewards(DEFAULT_BALANCE);
+        })
+        .catch((err) => {
+          throw err;
+        })
     );
+  } else {
+    const postData = {} as PostRetrieveReward;
+    const pendingRequestData = [] as PendingRequestRetrieveReward[];
 
-    multicallCollectRewards.push(mcr);
-  });
+    uniqueRoundIds.forEach((roundId) => {
+      data.forEach((datum) => {
+        if (datum.roundId === roundId) {
+          const pendingRequest = {} as PendingRequestRetrieveReward;
+          postData.roundId = datum.roundId;
+          postData.voterAddress = datum.address;
+          pendingRequest.ancillaryData = datum.ancillaryData
+            ? datum.ancillaryData
+            : "0x";
+          pendingRequest.identifier = ethers.utils.toUtf8Bytes(
+            datum.identifier
+          );
+          pendingRequest.time = datum.time;
+          pendingRequestData.push(pendingRequest);
+        }
+      });
```

Comment:
nit: seems like this might be a bit simpler -- I think the looping structure isn't as necessary here as it is with the multicall:
```ts
    const [roundId] = uniqueRoundIds;
    const postData.pendingRequests = data.filter((datum) => datum.roundId === roundId)
      .map((datum): PendingRequestRetrieveReward  => ({
        ...datum,
        ancillaryData: datum.ancillaryData ? datum.ancillaryData : "0x",
        identifier: ethers.utils.toUtf8Bytes(datum.identifier)
      })
    );

Reply:
If you don't love the chained calls (they aren't super readable), you could do something more like:
```ts
const filteredRequests = data.filter((datum) => datum.roundId === roundId);
postData.pendingRequests = filteredRequests.map(...);
```

Reply:
Yup, that works great.